From 5ba5a13e5acd8c3a9194357be0120e6e8dd43ff8 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Thu, 6 Apr 2023 02:38:06 +0200
Subject: [PATCH] Revert "fix rendering of the infinite ocean effect"

This reverts https://github.com/fabiangreffrath/crispy-doom/commit/5fd8c7627adffdad220afd8fd7371bdf7a5d3eb2.

Having this patch in for some reason makes the game hang when
you start a new game.
---
 src/doom/p_setup.c | 18 ------------------
 src/doom/r_segs.c  |  4 ++--
 2 files changed, 2 insertions(+), 20 deletions(-)

diff --git a/src/doom/p_setup.c b/src/doom/p_setup.c
index 17d8461..a1893d4 100644
--- a/src/doom/p_setup.c
+++ b/src/doom/p_setup.c
@@ -277,18 +277,6 @@ void P_LoadSegs (int lump)
 }
 
 // [crispy] fix long wall wobble
-
-static angle_t anglediff(angle_t a, angle_t b)
-{
-	if (b > a)
-		return anglediff(b, a);
-
-	if (a - b < ANG180)
-		return a - b;
-	else // [crispy] wrap around
-		return b - a;
-}
-
 void P_SegLengths (boolean contrast_only)
 {
     int i;
@@ -310,12 +298,6 @@ void P_SegLengths (boolean contrast_only)
 		viewx = li->v1->r_x;
 		viewy = li->v1->r_y;
 		li->r_angle = R_PointToAngleCrispy(li->v2->r_x, li->v2->r_y);
-		// [crispy] more than just a little adjustment?
-		// back to the original angle then
-		if (anglediff(li->r_angle, li->angle) > ANG60/2)
-		{
-			li->r_angle = li->angle;
-		}
 	}
 
 	// [crispy] smoother fake contrast
diff --git a/src/doom/r_segs.c b/src/doom/r_segs.c
index 6b883a2..6c74c7b 100644
--- a/src/doom/r_segs.c
+++ b/src/doom/r_segs.c
@@ -501,8 +501,8 @@ void R_RenderSegLoop (void)
 // above R_StoreWallRange
 fixed_t R_ScaleFromGlobalAngle (angle_t visangle)
 {
-    angle_t	anglea = ANG90 + (visangle - viewangle);
-    angle_t	angleb = ANG90 + (visangle - rw_normalangle);
+    int		anglea = ANG90 + (visangle - viewangle);
+    int		angleb = ANG90 + (visangle - rw_normalangle);
     int		den = FixedMul(rw_distance, finesine[anglea >> ANGLETOFINESHIFT]);
     fixed_t	num = FixedMul(projection, finesine[angleb >> ANGLETOFINESHIFT])<<detailshift;
     fixed_t 	scale;
-- 
2.39.0

